artiq/coredevice/mirny.py

```diff
@@ -78,15 +98,16 @@ def init(self, blind=False):
 
         :param blind: Do not attempt to verify presence and compatibility.
         """
+        reg0 = self.read_reg(0)
+        self.hw_rev = reg0 & 0x3
+
         if not blind:
-            reg0 = self.read_reg(0)
-            if reg0 & 0b11 != 0b11:
-                raise ValueError("Mirny HW_REV mismatch")
-            if (reg0 >> 2) & 0b11 != 0b00:
+            if (reg0 >> 2) & 0x3 != PROTO_REV_MATCH:
                 raise ValueError("Mirny PROTO_REV mismatch")
             delay(100 * us)  # slack
 
         # select clock source
+        self.clk_sel = self.clk_sel_hw_rev[self.hw_rev - 2]
```

Comment:
With `hw_rev` in `[0, 1, 2, 3]`, this is `[-2, -1, 0, 1]`, i.e. `[0, 1, 0, 1]` silently wrapping around. Does this make sense? I'd rather have `clk_sel_hw_rev` be a full 4-element array (with error values where needed).

Reply:
Does d691b05 address this appropriately?

Reply:
Yes. Thanks.